### PR TITLE
Use the extract api to load losses_by_asset as layer

### PR DESCRIPTION
**Pull request by ptormene**
Companion of https://github.com/gem/oq-engine/pull/3521

One step towards https://github.com/gem/oq-irmt-qgis/issues/401

### CODE REVIEW

**Comment by micheles** [comment]
LGTM if the tests are green

**Comment by ptormene** [comment]
Fixes https://github.com/gem/oq-irmt-qgis/issues/405

File: svir/test/integration/test_drive_oq_engine.py [hunk 1]
```diff
@@ -171,12 +174,90 @@ def load_output(self, calc, output):
             dlg = OUTPUT_TYPE_LOADERS[output_type](
                 IFACE, Mock(), self.session, self.hostname, calc_id,
                 output_type, filepath)
+            if dlg.ok_button.isEnabled():
+                dlg.accept()
+                print('\t\tok')
```
Comment (micheles) [inline svir/test/integration/test_drive_oq_engine.py]: Why is this print needed?

Comment (micheles) [reply svir/test/integration/test_drive_oq_engine.py]: Is it printed when running the tests? If yes, please add a comment saying so.

File: svir/test/integration/test_drive_oq_engine.py [hunk 2]
```diff
@@ -55,8 +57,17 @@ def setUp(self):
         self.session = Session()
         self.hostname = os.environ.get('OQ_ENGINE_HOST',
                                        'http://localhost:8800')
+        self.reset_gui()
+
+    def tearDown(self):
+        del self.session
+        del self.hostname
+        del self.viewer_dock
+
```
Comment (micheles) [inline svir/test/integration/test_drive_oq_engine.py]: Is this the part solving the segfault? If so, maybe a comment is in order.

File: svir/dialogs/viewer_dock.py [hunk 3]
```diff
@@ -1162,23 +1162,33 @@ def clear_plot(self):
...
-            self.imt_cbx.clear()
-            self.imt_cbx.blockSignals(False)
+            try:
+                self.imt_cbx.blockSignals(True)
+                self.imt_cbx.clear()
+                self.imt_cbx.blockSignals(False)
+            except RuntimeError:
+                # continue if something like this occurs:
+                # "wrapped C/C++ object of type QComboBox has been deleted"
+                pass
```
Comment (micheles) [inline svir/dialogs/viewer_dock.py]: I would at least log the error, it is scary to hide all the RuntimeErrors

File: svir/dialogs/viewer_dock.py [hunk 4]
```diff
@@ -1162,23 +1162,33 @@ def clear_plot(self):
...
-            self.loss_type_cbx.clear()
-            self.loss_type_cbx.blockSignals(False)
+            try:
+                self.loss_type_cbx.blockSignals(True)
+                self.loss_type_cbx.clear()
+                self.loss_type_cbx.blockSignals(False)
+            except RuntimeError:
+                # continue if something like this occurs:
+                # "wrapped C/C++ object of type QComboBox has been deleted"
+                pass
```
Comment (micheles) [inline svir/dialogs/viewer_dock.py]: Same as above

File: svir/test/integration/test_drive_oq_engine.py (hunk 1 above)
Comment (ptormene) [reply svir/test/integration/test_drive_oq_engine.py]: From the context, I suppose it is already clear enough. There are many prints of that kind in the workflow, like "Loading output X... ok"

File: svir/test/integration/test_drive_oq_engine.py (hunk 2 above)
Comment (ptormene) [reply svir/test/integration/test_drive_oq_engine.py]: The segfault is solved by another thing. I've just added a comment

File: svir/dialogs/viewer_dock.py (hunk 3 above)
Comment (ptormene) [reply svir/dialogs/viewer_dock.py]: Done

File: svir/dialogs/viewer_dock.py (hunk 4 above)
Comment (ptormene) [reply svir/dialogs/viewer_dock.py]: Done

**Pull request merged**